src/compiler/emitter.ts

```diff
@@ -379,6 +379,93 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
             return true;
         }
 
+        const enum Jump {
+            Break       = 1 << 1,
+            Continue    = 1 << 2,
+            Return      = 1 << 3
+        }
+
+        interface ConvertedLoopState {
+            // set of labels that occured inside the converted loop
+            // used to determine if labeled jump can be emitted as is or it should be dispatched to calling code
+            labels?: Map<string>;
+            // collection of labeled jumps that transfer control outside the converted loop.
+            // maps store association 'label -> labelMarker' where
+            // - label - value of label as it apprear in code
+            // - label marker - return value that should be interpreted by calling code as 'jump to <label>' 
+            labeledNonLocalBreaks?: Map<string>;
+            labeledNonLocalContinues?: Map<string>;
+
+            // set of non-labeled jumps that transfer control outside the converted loop
+            // used to emit dispatching logic in the caller of converted loop
+            nonLocalJumps?: Jump;
+
+            // set of non-labeled jumps that should be interpreted as local
+            // i.e. if converted loop contains normal loop or switch statement then inside this loop break should be treated as local jump
+            allowedNonLabeledJumps?: Jump;
+
+            // alias for 'arguments' object from the calling code stack frame
+            // i.e.
+            // for (let x;;) <statement that captures x in closure and uses 'arguments'>
+            // should be converted to
+            // var loop = function(x) { <code where 'arguments' is replaced witg 'arguments_1'> }
+            // var arguments_1 = arguments
+            // for (var x;;) loop(x);
+            // otherwise semantics of the code will be different since 'arguments' inside converted loop body 
+            // will refer to function that holds converted loop.
+            // This value is set on demand.
+            argumentsName?: string;

```

Comment:
I don't see an example of `arguments` being used both inside and outside of the loop. Can you point it out to me or add one?